models/classes/user/Lti1p3User.php

```diff
@@ -38,11 +38,13 @@ class Lti1p3User extends LtiUser
      * @throws \common_exception_Error
      * @throws LtiVariableMissingException
      */
-    public function __construct($launchData)
+    public function __construct($launchData, string $userUri = null)
     {
-        $userUri = $launchData->hasVariable(LtiLaunchData::USER_ID)
-            ? $launchData->getVariable(LtiLaunchData::USER_ID)
-            : self::ANONYMOUS_USER_URI;
+        if ($userUri === null) {
```

Comment:
Ok you will probably hate me but $userUri is already defined by $launchData so we can change behavior of $lanchData that it will return $userUri you want rather then changing this Lti1p3User class. Wdyt?

Reply:
The thing is that this info is not part of the lti token but part of lti user. Tha's why I left it here.

Reply:
Yeah I know but it could look better, its just suggestion.